Gemfile

```diff
@@ -28,31 +46,60 @@ group :system_tests do
   gem 'beaker-puppet_install_helper', :require => false
 end
 
-is_x64 = Gem::Platform.local.cpu == 'x64'
-if is_x64
-  platform(:x64_mingw) do
-    gem "win32-dir", "~> 0.4.9", :require => false
-    gem "win32-process", "~> 0.7.4", :require => false
-    gem "win32-service", "~> 0.8.4", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-  end
-else
-  platform(:mingw) do
-    gem "win32-process", "~> 0.6.5", :require => false
-    gem "win32-service", "~> 0.7.2", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-    if RUBY_VERSION =~ /^1\./ then
-      gem "win32console", :require => false
+# The recommendation is for PROJECT_GEM_VERSION,
+# there are older ways of referencing these. Add
+# them all for compatibility reasons. Remove
+# later when no issues are known
+puppetversion = ENV['PUPPET_GEM_VERSION'] || ENV['GEM_PUPPET_VERSION'] || ENV['PUPPET_LOCATION'] || '>= 0'
+gem 'puppet', *location_for(puppetversion)
+
+gem "facter", *location_for(ENV['FACTER_GEM_VERSION'] || ENV['GEM_FACTER_VERSION'] || ENV['FACTER_LOCATION'] || '>= 0')
+gem "hiera", *location_for(ENV['HIERA_GEM_VERSION'] || ENV['GEM_HIERA_VERSION'] || ENV['HIERA_LOCATION'] || '>= 0')
```

Comment:
So the explicit inclusion of Hiera here is what runs afoul in Travis.  I turned on `DEBUG_RESOLVER=1` in a separate PR and I can see that Bundler has trouble with resolving Hiera from Puppet 3.8.1 all the way down, until Puppet 3.0.0, at which point it finally matches on a Hiera version - see https://api.travis-ci.org/jobs/69829416/log.txt?deansi=true

So yeah, Bundler--